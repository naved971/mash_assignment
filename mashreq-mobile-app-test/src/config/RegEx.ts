export const RegEx : any = {
    USERNAME:/^[a-zA-Z0-9]{10,}$/,
    USERNAME_UAE: /^[a-zA-Z0-9]{5,}$/,
    USERNAME_INDIA: /^[a-zA-Z][a-zA-Z0-9]{5,}$/,
    USERNAME_SAUDIA: /^[a-zA-Z0-9]{8,}$/,
    USERNAME_EGYPT: /^[a-zA-Z]{10,}$/,
    PASSWORD: /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/,
    ALPHABETS_ONLY_2CHARS: /^[a-zA-Z]{2,}$/

};