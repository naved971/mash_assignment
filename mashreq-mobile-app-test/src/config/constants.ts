export const USERNAME_KEY = 'username';


export enum Languages {
    en = 'en',
    ja = 'ja',
    nl = 'nl'
}
export enum GlobalErrors {
    NetworkError = "Network Error"
}

export enum APIStatus {
    Error = "error",
    Success = "success"
}

export enum Countries {
    UAE = "UAE",
    India = "India",
    Egypt = "Egypt",
    Saudia = "Saudia"
}

export const CountryImages: any = {
    [Countries.UAE]: require('../assets/UAE.png'),
    [Countries.India]: require('../assets/India.png'),
    [Countries.Egypt]: require('../assets/Egypt.png'),
    [Countries.Saudia]: require('../assets/Saudia.png'),
}

export const LanguagesImages: any = {
    [Languages.en]: require('../assets/en.png'),
    [Languages.ja]: require('../assets/ja.png'),
    [Languages.nl]: require('../assets/nl.png'),
}