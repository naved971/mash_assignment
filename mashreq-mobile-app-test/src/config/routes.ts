export const ROUTE_USER_REGISTRATION = 'user_registration';
export const ROUTE_USER_DASHBOARD = 'user_dashboard';
