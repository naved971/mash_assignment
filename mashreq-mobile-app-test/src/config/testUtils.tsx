import React, { PropsWithChildren } from 'react';
import { ThemeProvider } from 'styled-components/native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux'
import { setupStore } from '../redux/store'
import { render } from '@testing-library/react-native';


export const mockTheme = {
  // Your theme object goes here
  TextPrime: '#3d3d3d',
  ErrorTextAlert: '#cc0000',
  LightTextPrime: '#797979',
  SubtleHuePrimary: '#FFF3E0',
  LuminousPrimary: '#FFB74D',
  InactiveShade: '#D3D3D3',
  CoreChroma: "#FF9800",
  TextSecondary: "#ffffff"
};


export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: any = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
          <NavigationContainer>
            <ThemeProvider theme={mockTheme}>
              {children}
            </ThemeProvider>
          </NavigationContainer>
      </Provider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}



