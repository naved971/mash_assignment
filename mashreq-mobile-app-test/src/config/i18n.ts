/* eslint-disable */

import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import english from '../localization/en.json'
import dutch from '../localization/nl.json'
import japanese from '../localization/ja.json'
import { Languages } from './constants'

export const resources = {
  [Languages.en]: {
    translation: {
      ...english,
    }
  },
  [Languages.ja]: {
    translation: {
      ...japanese,
    }
  },
  [Languages.nl]: {
    translation: {
      ...dutch
    }
  }
}

i18n.use(initReactI18next).init({
  compatibilityJSON: 'v3',
  resources,
  lng: 'en',
  fallbackLng: 'en',
  interpolation: {
    escapeValue: false
  }
})

export default i18n
