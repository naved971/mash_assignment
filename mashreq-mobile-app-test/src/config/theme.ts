import { Countries } from "./constants";

export const base = {
    BackgroundCore: "#ffffff",
    LightBackgroundCore: "#f7f7f7",
    SecondaryBackdrop: "#3d3d3d",
    LightSecondaryBackdrop: "#797979",
    TextPrime: "#3d3d3d",
    LightTextPrime: "#797979",
    TextSecondary: "#ffffff",
    TextBackgroundPrime: "#ffffff",
    TextBackgroundSecondary: "#3d3d3d",
    InactiveShade: '#D3D3D3',
    ErrorTextAlert: '#cc0000'
};

const defaultPalettes =  {
    SubtleHuePrimary: "#E0F2F1",
    LuminousPrimary: "#4DB6AC",
    CoreChroma: "#009688",
    VibrantCore: "#00695C",
    ForegroundPrime: "#FFFFFF"
}

export const themePalettes = {
    [Countries.UAE]: {
        SubtleHuePrimary: "#FFDAB9",
        LuminousPrimary: "#FFA07A",
        CoreChroma: "#FF4500",
        VibrantCore: "#FF8C00",
        ForegroundPrime: "#FFFFFF"
    },
    [Countries.India]: {
        SubtleHuePrimary: "#FFE5B4",
        LuminousPrimary: "#FFFFFF",
        CoreChroma: "#138808",
        VibrantCore: "#FF4500",
        ForegroundPrime: "#000000"
    },
    [Countries.Egypt]: {
        SubtleHuePrimary: "#E6E6E6",
        LuminousPrimary: "#FFFFFF",
        CoreChroma: "#C8102E",
        VibrantCore: "#000000",
        ForegroundPrime: "#C8102E"
    },
    [Countries.Saudia]: {
        SubtleHuePrimary: "#DFF0D8",
        LuminousPrimary: "#FFFFFF",
        CoreChroma: "#008033",
        VibrantCore: "#154734",
        ForegroundPrime: "#000000"
    },
    default:defaultPalettes
};