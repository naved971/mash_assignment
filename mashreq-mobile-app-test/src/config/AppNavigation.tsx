import React from 'react';
import { NavigationContainer, DefaultTheme } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import RegisterUser from '../features/registerUser';
import Dashboard from '../features/dashboard';
import * as Routes from './routes';
import { ThemeProvider } from 'styled-components/native';
import { SafeAreaView, StyleSheet } from 'react-native';
import { useSelector } from 'react-redux';

const Stack = createNativeStackNavigator();

const AppNavigation = () => {
  const { theme } = useSelector((state: any) => state.themeReducer);
  const styles = useStyles(theme);

  const appTheme = {
    ...DefaultTheme,
    colors: {
      ...DefaultTheme.colors,
      background: theme.BackgroundCore,
    },
  };

  return (
    <ThemeProvider theme={theme}>
      <SafeAreaView style={styles.safeAreaContainer}>
        <NavigationContainer theme={appTheme}>
          <Stack.Navigator
            screenOptions={{
              headerShown: false,
            }}
          >
            <Stack.Screen name={Routes.ROUTE_USER_REGISTRATION} component={RegisterUser} />
            <Stack.Screen name={Routes.ROUTE_USER_DASHBOARD} component={Dashboard} />
          </Stack.Navigator>
        </NavigationContainer>
      </SafeAreaView>
    </ThemeProvider>
  );
};

const useStyles = (theme: any) =>
  StyleSheet.create({
    safeAreaContainer: {
      flex: 1,
      backgroundColor: theme.BackgroundCore,
    },
  });

export default AppNavigation;
