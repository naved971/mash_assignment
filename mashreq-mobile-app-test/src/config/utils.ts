
import i18n from 'i18next'
import { I18nManager } from 'react-native';
import RNRestart from 'react-native-restart';
import { base, themePalettes } from "../config/theme";
import { Countries, GlobalErrors } from './constants';
import { AxiosError, AxiosResponse } from 'axios';
import SecureStore, { ACCESSIBLE } from "react-native-secure-key-store";

export const setLanguage = async (languageKey: string) => {
  const isNewLanguageRTL = i18n.dir(languageKey) === 'rtl';

  const isCurrentLayoutRTL = I18nManager.isRTL;

  const isLayoutChangeNeeded = isCurrentLayoutRTL !== isNewLanguageRTL;

  i18n.changeLanguage(languageKey)
  if (isLayoutChangeNeeded) {
    I18nManager.forceRTL(isNewLanguageRTL); // successful, since otherwise, the new
    RNRestart.restart();                    // language would not apply after restart.
  }
};

export const getTheme = (country?: string) => {
  switch (country) {
    case Countries.UAE:
      return { ...base, ...themePalettes[Countries.UAE] }
    case Countries.India:
      return { ...base, ...themePalettes[Countries.India] }
    case Countries.Egypt:
      return { ...base, ...themePalettes[Countries.Egypt] }
    case Countries.Saudia:
      return { ...base, ...themePalettes[Countries.Saudia] }
    default:
      return { ...base, ...themePalettes.default };
  }
}

export const handleNetworkError = (error: AxiosError) => {
  const axiosError = error as AxiosError;
  const response: AxiosResponse<any> | undefined = axiosError.response;
  const errorMessage = response?.data?.message || GlobalErrors.NetworkError;
  return errorMessage;
}


export const setKeyStore = async (key: string, value: string): Promise<void> => {
  try {
    await SecureStore.set(key, value, { accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY });
  } catch (error) {
    throw error;
  }
};

// Function to retrieve the username securely
export const getKeyStore = async (key: string): Promise<string | null> => {
  try {
    const value = await SecureStore.get(key);
    return value || null;
  } catch (error) {
    throw error;
  }
};

// Function to delete the saved username
export const deleteKeyStore = async (key: string): Promise<void> => {
  try {
    await SecureStore.remove(key);
  } catch (error) {
    throw error;
  }
};

