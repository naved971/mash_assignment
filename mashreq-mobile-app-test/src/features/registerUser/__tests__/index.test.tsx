import React from 'react';
import RegisterUser from '../index';
import { mockTheme, renderWithProviders } from '../../../config/testUtils';
import { fireEvent } from '@testing-library/react-native';
import { useNavigation } from '@react-navigation/native'
import * as Utils from '../../../config/utils';

const mockState = {
    userRegistration: {
        response: { status: "success" },
        loading: false,
        error: null,
    }
};

jest.mock('styled-components', () => ({
    ...jest.requireActual('styled-components'),
    useTheme: () => (mockTheme),
}));


jest.mock('react-i18next', () => ({
    // mock useTranslation hook 
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: jest.fn(),
                language: 'EN'
            }
        }
    }
}))

jest.mock('i18next', () => ({
    changeLanguage: jest.fn(),
    dir: jest.fn(),
}))

jest.mock('react-redux', () => {
    return {
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn().mockImplementation((selectorFn) => selectorFn(mockState)),
        useDispatch: () => jest.fn(),
    };
});

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: ''
}))

jest.mock('react-native-keyboard-aware-scroll-view', () => {
    const KeyboardAwareScrollView = ({ children }: any) => children;
    return { KeyboardAwareScrollView };
});

jest.mock('@react-navigation/native', () => ({
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: jest.fn(),
}));

 // Mock the functions from '../../config/utils'
jest.mock('../../../config/utils', () => ({
    setKeyStore: jest.fn(),
    getTheme: jest.fn(),  // Mock the getTheme function
    setLanguage : jest.fn()
}));

const mockedUtils = Utils as jest.Mocked<typeof Utils>;


test('renders RegisterUser component', async () => {

    const mockNavigation = {
        navigate: jest.fn(),
        dispatch: jest.fn(),
    };
    (useNavigation as jest.Mock).mockReturnValue(mockNavigation);

    mockedUtils.setKeyStore.mockResolvedValue();

    // Render the component with mocked dependencies
    const { toJSON , getByTestId } = renderWithProviders(<RegisterUser/>);
    fireEvent.press(getByTestId('languageButton')); 
    fireEvent.press(getByTestId('selectContainer')); 
    fireEvent.press(getByTestId('languageListItem_0')); 
    fireEvent.press(getByTestId('countryListItem_0')); 


    const usernameInput = getByTestId('username');
    // Simulate user input in the username field
    fireEvent.changeText(usernameInput, 'testUsername');

    const passwordInput = getByTestId('password');
    // Simulate user input in the password field
    fireEvent.changeText(passwordInput, 'testUser@123');

    const firstNameInput = getByTestId('firstName');
    // Simulate user input in the firstName field
    fireEvent.changeText(firstNameInput, 'naved');

    const lastNameInput = getByTestId('lastName');
    // Simulate user input in the lastName field
    fireEvent.changeText(lastNameInput, 'khan');

    // Expect the component to match the snapshot
    expect(toJSON()).toMatchSnapshot();
});