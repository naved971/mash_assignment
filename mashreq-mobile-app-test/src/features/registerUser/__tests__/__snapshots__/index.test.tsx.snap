// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RegisterUser component 1`] = `
<View
  style={
    {
      "flex": 1,
      "paddingHorizontal": 22,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-end",
          "marginBottom": 30,
          "padding": 22,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
        testID="languageButton"
      >
        <
          color="#FF9800"
          icon={
            {
              "icon": [
                512,
                512,
                [
                  127760,
                ],
                "f0ac",
                "M352 256c0 22.2-1.2 43.6-3.3 64H163.3c-2.2-20.4-3.3-41.8-3.3-64s1.2-43.6 3.3-64H348.7c2.2 20.4 3.3 41.8 3.3 64zm28.8-64H503.9c5.3 20.5 8.1 41.9 8.1 64s-2.8 43.5-8.1 64H380.8c2.1-20.6 3.2-42 3.2-64s-1.1-43.4-3.2-64zm112.6-32H376.7c-10-63.9-29.8-117.4-55.3-151.6c78.3 20.7 142 77.5 171.9 151.6zm-149.1 0H167.7c6.1-36.4 15.5-68.6 27-94.7c10.5-23.6 22.2-40.7 33.5-51.5C239.4 3.2 248.7 0 256 0s16.6 3.2 27.8 13.8c11.3 10.8 23 27.9 33.5 51.5c11.6 26 20.9 58.2 27 94.7zm-209 0H18.6C48.6 85.9 112.2 29.1 190.6 8.4C165.1 42.6 145.3 96.1 135.3 160zM8.1 192H131.2c-2.1 20.6-3.2 42-3.2 64s1.1 43.4 3.2 64H8.1C2.8 299.5 0 278.1 0 256s2.8-43.5 8.1-64zM194.7 446.6c-11.6-26-20.9-58.2-27-94.6H344.3c-6.1 36.4-15.5 68.6-27 94.6c-10.5 23.6-22.2 40.7-33.5 51.5C272.6 508.8 263.3 512 256 512s-16.6-3.2-27.8-13.8c-11.3-10.8-23-27.9-33.5-51.5zM135.3 352c10 63.9 29.8 117.4 55.3 151.6C112.2 482.9 48.6 426.1 18.6 352H135.3zm358.1 0c-30 74.1-93.6 130.9-171.9 151.6c25.5-34.2 45.2-87.7 55.3-151.6H493.4z",
              ],
              "iconName": "globe",
              "prefix": "fas",
            }
          }
          size={24}
        />
        <View
          style={
            {
              "marginLeft": 6,
            }
          }
        >
          <Text
            color="#FF9800"
            style={
              [
                {
                  "color": "#FF9800",
                },
                {
                  "fontSize": 16,
                  "fontWeight": "600",
                },
              ]
            }
          >
            EN
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        color="#3d3d3d"
        style={
          [
            {
              "color": "#3d3d3d",
            },
            {
              "fontSize": 22,
              "fontWeight": "bold",
            },
          ]
        }
      >
        title
      </Text>
      <View
        style={
          {
            "paddingVertical": 20,
          }
        }
      >
        <Text
          color="#FF9800"
          style={
            [
              {
                "color": "#FF9800",
              },
              {
                "fontSize": 18,
                "fontWeight": "600",
              },
            ]
          }
        >
          signup
        </Text>
      </View>
    </View>
    <Text
      color="#3d3d3d"
      style={
        [
          {
            "color": "#3d3d3d",
          },
          {
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      countryLabel
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderColor": "#3d3d3d",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 54,
          "marginVertical": 12,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
          "width": "100%",
        }
      }
      testID="selectContainer"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 8,
            "width": 30,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/UAE.png",
            }
          }
          style={
            {
              "height": 24,
              "marginHorizontal": 6,
              "resizeMode": "contain",
              "width": 24,
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <Text
          color="#3d3d3d"
          style={
            [
              {
                "color": "#3d3d3d",
              },
              {
                "fontSize": 10,
                "fontWeight": "400",
              },
            ]
          }
        >
          chooseCountry
        </Text>
        <Text
          color="#3d3d3d"
          style={
            [
              {
                "color": "#3d3d3d",
              },
              {
                "fontSize": 16,
                "fontWeight": "400",
              },
            ]
          }
        >
          UAE
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 8,
            "width": 30,
          }
        }
        testID="iconContainer"
      >
        <
          color="#3d3d3d"
          icon={
            {
              "icon": [
                320,
                512,
                [
                  8250,
                ],
                "f105",
                "M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z",
              ],
              "iconName": "angle-right",
              "prefix": "fas",
            }
          }
          size={18}
        />
      </View>
    </View>
    <View
      style={
        {
          "marginVertical": 12,
        }
      }
    >
      <TextInput
        keyboardType="default"
        maxLength={16}
        onChangeText={[Function]}
        placeholder="username"
        placeholderTextColor="#3d3d3d"
        style={
          {
            "alignItems": "center",
            "borderColor": "#3d3d3d",
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 54,
            "padding": 12,
            "width": "100%",
          }
        }
        testID="username"
        value="testUsername"
      />
      <View
        style={
          {
            "marginTop": 4,
          }
        }
      >
        <Text
          color="#797979"
          style={
            [
              {
                "color": "#797979",
              },
              {
                "fontSize": 12,
                "fontWeight": "400",
              },
            ]
          }
        >
          validations.usernameHelperUAE
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "marginVertical": 12,
        }
      }
    >
      <TextInput
        keyboardType="default"
        maxLength={16}
        onChangeText={[Function]}
        placeholder="password"
        placeholderTextColor="#3d3d3d"
        secureTextEntry={true}
        style={
          {
            "alignItems": "center",
            "borderColor": "#3d3d3d",
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 54,
            "padding": 12,
            "width": "100%",
          }
        }
        testID="password"
        value="testUser@123"
      />
    </View>
    <View
      style={
        {
          "marginVertical": 12,
        }
      }
    >
      <TextInput
        keyboardType="default"
        maxLength={16}
        onChangeText={[Function]}
        placeholder="firstName"
        placeholderTextColor="#3d3d3d"
        style={
          {
            "alignItems": "center",
            "borderColor": "#3d3d3d",
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 54,
            "padding": 12,
            "width": "100%",
          }
        }
        testID="firstName"
        value="Naved"
      />
    </View>
    <View
      style={
        {
          "marginVertical": 12,
        }
      }
    >
      <TextInput
        keyboardType="default"
        maxLength={16}
        onChangeText={[Function]}
        placeholder="lastName"
        placeholderTextColor="#3d3d3d"
        style={
          {
            "alignItems": "center",
            "borderColor": "#3d3d3d",
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 54,
            "padding": 12,
            "width": "100%",
          }
        }
        testID="lastName"
        value="khan"
      />
    </View>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FF9800",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "height": 54,
          "justifyContent": "center",
          "marginVertical": 8,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        }
      }
      testID="registerButton"
    >
      <Text
        color="#ffffff"
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        register
      </Text>
    </View>
  </View>
</View>
`;
